packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol

```diff
@@ -1,47 +1,235 @@
+     * @param recipient L1 address that should receive the tokens.
+     * @param l2Token L2 token to deposit.
+     * @param amount How many L2 tokens should be deposited.
+     * @param maxFee Max fraction of the total `amount` that the depositor is willing to pay as a fee. scaled by 1e18.
+     */
     function deposit(
         address recipient,
-        address l1Token,
+        address l2Token,
         uint256 amount,
         uint256 maxFee
-    ) public {}
+    ) public onlyIfDepositsEnabled() {
+        require(isWhitelistToken(l2Token), "deposit token not whitelisted");
+        require(maxFee <= 1e18, "max fee can not be over 1e18");
+
+        emit FundsDeposited(
+            numberOfDeposits, // the current number of deposits acts as a deposit ID (nonce).
+            getCurrentTime(),
+            msg.sender,
+            recipient,
+            whitelistedTokens[l2Token].l1Token,
+            amount,
+            maxFee
+        );
+
+        numberOfDeposits += 1;
+
+        TokenHelper.safeTransferFrom(l2Token, msg.sender, address(this), amount);
+    }
+
+    /**************************************
+     *          RELAYER FUNCTIONS         *
+     **************************************/
+
+    /**
+     * @notice Called by relayer (or any other EOA) to move a batch of funds from the deposit box, through the canonical
+     *      token bridge, to the L1 Withdraw box.
+     * @dev The frequency that this function can be called is rate limited by the `minimumBridgingDelay` to prevent spam
+     *      on L1 as the finalization of a L2->L1 tx is quite expensive.
+     * @param l2Token L2 token to relay over the canonical bridge.
+     * @param l1Gas Unused by optimism, but included for potential forward compatibility considerations.
+     */
+    function bridgeTokens(address l2Token, uint32 l1Gas) public {
+        uint256 bridgeDepositBoxBalance = TokenLike(l2Token).balanceOf(address(this));
+        require(bridgeDepositBoxBalance > 0, "can't bridge zero tokens");
+        require(isWhitelistToken(l2Token), "can't bridge non-whitelisted token");
+        require(hasEnoughTimeElapsedToBridge(l2Token), "not enough time has elapsed from previous bridge");
+
+        StandardBridgeLike(Lib_PredeployAddresses.L2_STANDARD_BRIDGE).withdrawTo(
+            l2Token, // _l2Token. Address of the L2 token to bridge over.
+            l1WithdrawContract, // _to. Withdraw, over the bridge, to the l1Withdraw contract.
+            bridgeDepositBoxBalance, // _amount. Send the full balance of the deposit box to bridge.
+            l1Gas, // _l1Gas. Unused, but included for potential forward compatibility considerations
+            "0x" // _data. TODO: add additional info into this data prop this.
+        );
+
+        emit TokensBridged(l2Token, bridgeDepositBoxBalance, l1Gas, msg.sender);
+        whitelistedTokens[l2Token].lastBridgeTime = uint64(getCurrentTime());
```

Comment:
Should be making external contract call to `StandardBridgeLike` after all state modifications are done.

Reply:
good point. done.